### PR TITLE
Added handling of Meta node

### PR DESCRIPTION
**Pull request by fabricehong**
I implemented it the same way relationships are handled. That is to say : 
- if meta node is found in json but @meta node is present, ignore and don't throw error
- if @meta node is present but meta node in json is not present, ignore and don't throw error

however, I think it would be good to manage those situations in the future, in order to avoid debugging nightmares

### CODE REVIEW

**Comment by jasminb** [comment]
Please add unit tests to cover all cases regarding meta.

Another thing I've noticed is formatting:

```
if (a>b) -> if (a > b); # Spaces are missing.
```

**Comment by fabricehong** [comment]
done

**Comment by fabricehong** [comment]
Actually.. I'm not sure what I've done is correct because it changes the structure of the object type. Should we collect meta information with an hashmap ?

Still new to json API. But I realize that meta information might be different in each requests.

what do you think ?

**Comment by fabricehong** [comment]
Another possibility is to inherit from a jonapi object that manage common jsonapi attributes. It would also allow to avoid repeating mandatory attribute like @Id.

**Comment by jasminb** [comment]
As per specification, `meta` object can be anything.

So one solution around this would be to force user to define a specific object that reflects the expected meta for a given resource, another would be to have a `base` class that handles meta as a `map`.

I am more for explicit solution (what you already did) and that is, if you care about the `meta` object, you should know what to expect and therefore you should be able to define a class that represents it.

In case `meta` is same for all resource paths, user can share the same `Meta` class for all objects.

**Comment by fabricehong** [comment]
Agreed. But what about having the possibility to use inheritance for annotations ? what would greatly improve the library's flexibility.
I will create an issue for this

**Comment by jasminb** [comment]
Another thing related to pull request, `@Meta` attribute should be ignored during serialisation.

**Comment by fabricehong** [comment]
done

File: src/main/java/com/github/jasminb/jsonapi/ResourceConverter.java [hunk 1]
```diff
@@ -453,10 +453,15 @@ private boolean isCollection(JsonNode source) {
 		// Perform initial conversion
 		ObjectNode attributesNode = objectMapper.valueToTree(object);
 
-		// Remove id and relationship fields
+		// Remove id, meta and relationship fields
 		Field idField = ID_MAP.get(object.getClass());
 		attributesNode.remove(idField.getName());
 
+		Field metaField = META_FIELD.get(object.getClass());
+		if (metaField!=null) {
```
Comment (jasminb) [inline src/main/java/com/github/jasminb/jsonapi/ResourceConverter.java]: Formatting :)

**Pull request merged**